packages/assistify-ai/server/lib/SmartiAdapter.js

```diff
@@ -233,27 +196,92 @@ export class SmartiAdapter {
 	}
 
 	/**
-	 * This method provides an implementation for a hook registering an asynchronously sent response from Smarti to RocketChat
+	 * At any point in time when the anylsis for a room has been done (onClose, onMessage), this analysisCompleted
+	 * Updates the mapping and notifies the room, in order to reload the Smarti result representation
 	 *
-	 * @param roomId
-	 * @param smartiConversationId
-	 * @param token
+	 * @param roomId - the analyzed room ID
+	 * @param conversationId - the according Smarti conversation ID
+	 * @param timestamp - the analysis timestamp
+	 * @param token - ???
 	 */
-	static analysisCompleted(roomId, smartiConversationId, token) {
+	static analysisCompleted(roomId, conversationId, timestamp, token) {
```

Comment:
Is it not like that we've always got the conversation at hand once this method is called? Then, we could (and should) get the timestamp via `_getLastConversationAnalyzedTime` and remove the `conversationId` and `timestamp` from the signature.